File: Arrays/Medium/167/167.py
from typing import List

class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        front = 0
        back = len(numbers) - 1
        while front < back:
            sum = numbers[front] + numbers[back] - target
            if sum < 0:
                front += 1
            elif sum > 0:
                back -= 1
            else:
                return front + 1, back + 1

File: Arrays/Medium/167/167.c
#include <stdio.h>
#include <stdlib.h>

int* twoSum(int* numbers, int numbersSize, int target, int* returnSize){
    int *f = numbers;
    int *b = numbers + numbersSize - 1;
    int *res = malloc(sizeof(int) * 2);
    *returnSize = 2;
    int sum;
    while(f < b){
        sum = *f + *b;
        if(sum == target){
            res[0] = f - numbers + 1;
            res[1] = b - numbers + 1;
            return res;
        }else if(sum > target){
            b--;
        }else{
            f++;
        }
    }
    return res;
}
